### PR TITLE
Add YARD documentation

### PR DESCRIPTION
**Pull request by bess**
Use Yard (https://yardoc.org/) to auto-generate code documentation for this application. Eventually, we'll publish those docs on github pages and be able to use them as a reference.

### CODE REVIEW

**Comment by bess** [comment]
Screen Shot 2020-06-03 at 1 48 41 PM

File: .yardopts [hunk 1]
```diff
@@ -0,0 +1 @@
+yardoc -o docs app/**/*.rb
```
Comment (maxkadel) [inline .yardopts:1]: Are there any of these .yardoc files we want to gitignore or Dockerignore at this point?

File: README.md [hunk 2]
```diff
@@ -1,65 +1,94 @@
...
+
+### Prerequisites
 
 - Download [Docker Desktop](https://www.docker.com/products/docker-desktop) and log in
 
-# Docker Development Setup
+### Docker Development Setup
 
-## If this is your first time working in this repo or the Dockerfile.base has been updated, (re)build the base service (dependencies, etc. that don't change often)
+- If this is your first time working in this repo or the Dockerfile.base has been updated, (re)build the base service (dependencies, etc. that don't change often)
```
Comment (maxkadel) [inline README.md:15]: This might be for a different PR, but should we say that you should _pull_ the base if it's your first time or if it's been changed by someone else? Rather than building it?

File: README.md [hunk 3]
```diff
@@ -1,65 +1,94 @@
...
 
-```bash
-  docker-compose build base
-```
+  ```bash
+      docker-compose build base
+  ```
 
-## If this is your first time working in this repo or the Dockerfile has been updated you will need to (re)build those services
+- If this is your first time working in this repo or the Dockerfile has been updated you will need to (re)build those services
```
Comment (maxkadel) [inline README.md:21]: Again, change to pull, unless you are the one who made the changes and want to build *and push* the changes to the Dockerhub repo?

File: README.md [hunk 4]
```diff
@@ -1,65 +1,94 @@
...
   ```bash
   docker-compose up
   ```
 
 - Access the blacklight app at `http://localhost:3000`
+
 - Access the solr instance at `http://localhost:8983`
+
 - Access the image instance at `http://localhost:8182`
+
```
Comment (maxkadel) [inline README.md:40]: Add management app at `http://localhost:3001` ?

File: .yardopts (hunk 1 above)
Comment (bess) [reply .yardopts:1]: No, I think we need these so that the docs will build in the right way to get picked up by github pages.
